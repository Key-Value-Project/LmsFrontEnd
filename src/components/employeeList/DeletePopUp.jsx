import React from 'react';
import deleteIcon from '../../assets/icons/deleteIcon.svg';

const DeletePopUp = ({ open, handleClose, handleDelete }) => {
  return (
    <>
      {open && (
        <>
          <div className="delete-popup" data-testid="delete-popup">
            <div className="delete-popup-content">
              <img className="delete__icon" src={deleteIcon} alt="delete" />

              <h2>Are you sure you want to delete?</h2>
              <p>This action will delete the book from the corresponding shelf and this action is irreversible.</p>
              <div className="delete-popup-buttons">
                <button onClick={handleDelete}>Yes</button>
                <button onClick={handleClose}>No</button>
              </div>
            </div>
          </div>
        </>
      )}
    </>
  );
};

export default DeletePopUp;
