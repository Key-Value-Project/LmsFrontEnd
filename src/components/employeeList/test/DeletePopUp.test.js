import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import DeletePopUp from '../DeletePopUp';

describe('DeletePopUp Component Tests', () => {
    const mockHandleClose = jest.fn();
    const mockHandleDelete = jest.fn();

    test("Component not rendered when 'open' prop is false", () => {
        const { queryByTestId } = render(<DeletePopUp open={false} handleClose={mockHandleClose} handleDelete={mockHandleDelete} />);
        const popup = queryByTestId('delete-popup');
        expect(popup).toBeNull();
    });

    test("Component rendered when 'open' prop is true", () => {
        const { getByTestId } = render(<DeletePopUp open={true} handleClose={mockHandleClose} handleDelete={mockHandleDelete} />);
        const popup = getByTestId('delete-popup');
        expect(popup).toBeTruthy();
    });

    test("Clicking 'Yes' button calls handleDelete function", () => {
        const { getByText } = render(<DeletePopUp open={true} handleClose={mockHandleClose} handleDelete={mockHandleDelete} />);
        fireEvent.click(getByText('Yes'));
        expect(mockHandleDelete).toHaveBeenCalledTimes(1);
    });

    test("Clicking 'No' button calls handleClose function", () => {
        const { getByText } = render(<DeletePopUp open={true} handleClose={mockHandleClose} handleDelete={mockHandleDelete} />);
        fireEvent.click(getByText('No'));
        expect(mockHandleClose).toHaveBeenCalledTimes(1);
    });

    test('Snapshot', () => {
        const { asFragment } = render(<DeletePopUp open={true} handleClose={mockHandleClose} handleDelete={mockHandleDelete} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
