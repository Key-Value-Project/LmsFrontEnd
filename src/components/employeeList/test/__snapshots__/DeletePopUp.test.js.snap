// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePopUp Component Tests Snapshot 1`] = `
<DocumentFragment>
  <div
    class="delete-popup"
    data-testid="delete-popup"
  >
    <div
      class="delete-popup-content"
    >
      <h2>
        Are you sure you want to delete?
      </h2>
      <div
        class="delete-popup-buttons"
      >
        <button>
          Yes
        </button>
        <button>
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
